### PR TITLE
Add utilities for home directory.

### LINKED ISSUE #8: Cross-OS home directory command
https://stackoverflow.com/questions/25498128/how-do-i-tell-where-the-users-home-directory-is-in-dart

### PR DESCRIPTION
**Pull request by matanlurey**
Closes #8.

This is more or less copied from `pub`.

### CODE REVIEW

**Comment by matanlurey** [comment]
🏓 @kevmoo

File: lib/src/home_directory.dart [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+/// May return `null` on some platforms without a directory for this purpose.
+/// Flutter users should consider using the
+/// [path_provider](https://pub.dartlang.org/packages/path_provider) package,
+/// which has support for Android and iOS.
+String appDataPath(String path) {
+  if (Platform.isWindows) {
+    return p.join(Platform.environment['APPDATA'], path);
+  }
+  if (homePath != null) {
+    return p.join(homePath, '.$path');
```
Comment (kevmoo) [inline lib/src/home_directory.dart]: This worries me.

Mac: why not ~/Library/Application support?

Also: adding a `dot` before a path String feels...weird...

Comment (matanlurey) [reply lib/src/home_directory.dart]: re: OSX, you might not have privileges to write there, you usually do to APPDATA.

I'm definitely open to other suggestions though!

**Comment by kevmoo** [comment]
Ask Jake? Nate?

**Comment by matanlurey** [comment]
Ping @natebosch @jakemac53 anybody like to review? :)

File: lib/src/home_directory.dart [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+/// Returns an application data path for the current user and operating system.
+///
+/// May return `null` on some platforms without a directory for this purpose.
+/// Flutter users should consider using the
+/// [path_provider](https://pub.dartlang.org/packages/path_provider) package,
+/// which has support for Android and iOS.
+String appDataPath(String path) {
+  if (Platform.isWindows) {
+    return p.join(Platform.environment['APPDATA'], path);
```
Comment (jakemac53) [inline lib/src/home_directory.dart]: looks like this is vista and later, might be good to call out that this doesn't support previous windows versions

File: lib/src/home_directory.dart (hunk 1 above)
Comment (jakemac53) [reply lib/src/home_directory.dart]: I have zero domain knowledge here, but all the links I found online suggest some path under `~/Library`, and most of them specifically call out `~/Library/Application Support`.

**Comment by matanlurey** [comment]
Close for now since it's not clear we need this.

**Pull request closed without merging**